tensorflow/lite/java/src/main/java/org/tensorflow/lite/Tensor.java

```diff
@@ -311,7 +311,13 @@ private void throwIfTypeIsIncompatible(Object o) {
       return;
     }
     DataType oType = dataTypeOf(o);
+
     if (oType != dtype) {
+      // INT8 and UINT8 have the same string name, "byte"
+      if (oType.toStringName().equals(dtype.toStringName())) {
```

Comment:
Thanks! Can you add a test case in DataTypeTest.java to verify this equality? You can basically just copy this exact logic and make sure it holds for int8/uint8.